java/tsfile/src/main/java/org/apache/tsfile/write/writer/TsFileIOWriter.java

```diff
@@ -131,6 +132,8 @@ public class TsFileIOWriter implements AutoCloseable {
 
   protected String encryptKey;
 
+  private final List<FlushChunkMetadataListener> flushListeners = new CopyOnWriteArrayList<>();
```

Comment:
Is it necessary to use copy-on-write? The operations on the same TsFile are mostly single-threaded.
In what scenario can `addFlushListener` and `flush()` be called concurrently?

Reply:
This is just in case. fixed